File: restore.php
<?php

$log_file = "/home/sangam/.gemini/antigravity/brain/e74474c6-1c56-4cc7-9d32-ded42a474304/.system_generated/logs/overview.txt";
$lines = file($log_file);
$restored = 0;

foreach ($lines as $line) {
    $data = json_decode($line, true);
    if ($data && isset($data['type']) && $data['type'] === 'PLANNER_RESPONSE' && isset($data['tool_calls'])) {
        foreach ($data['tool_calls'] as $call) {
            if (isset($call['name']) && $call['name'] === 'write_to_file') {
                $args = $call['args'] ?? [];
                $target = trim($args['TargetFile'] ?? '', '"');
                $content = $args['CodeContent'] ?? '';
                
                if (strpos($target, 'app/Filament/') !== false) {
                    if (strpos($content, '"') === 0) {
                        $content = json_decode($content);
                    }
                    
                    // Apply fixes for Filament v3
                    $content = str_replace(
                        'protected static ?string $navigationGroup', 
                        'protected static string|\UnitEnum|null $navigationGroup', 
                        $content
                    );
                    $content = str_replace(
                        'protected static ?string $navigationIcon', 
                        'protected static string|\Illuminate\Contracts\Support\Htmlable|null $navigationIcon', 
                        $content
                    );
                    
                    if (!is_dir(dirname($target))) {
                        mkdir(dirname($target), 0755, true);
                    }
                    file_put_contents($target, $content);
                    echo "Restored: $target\n";
                    $restored++;
                }
            }
        }
    }
}
echo "Total restored: $restored\n";

File: restore.py
import json
import os

log_file = "/home/sangam/.gemini/antigravity/brain/e74474c6-1c56-4cc7-9d32-ded42a474304/.system_generated/logs/overview.txt"
restored = 0
with open(log_file, "r") as f:
    for line in f:
        try:
            data = json.loads(line)
            if data.get("type") == "PLANNER_RESPONSE" and "tool_calls" in data:
                for call in data["tool_calls"]:
                    if call.get("name") == "write_to_file":
                        args = call.get("args", {})
                        target = args.get("TargetFile", "").strip("\"")
                        content = args.get("CodeContent", "")
                        if "app/Filament/" in target:
                            if content.startswith("\"") and content.endswith("\""):
                                content = json.loads(content)
                            
                            # Replace the type hint to fix the Filament issue!
                            content = content.replace("protected static ?string $navigationGroup", "protected static string|\UnitEnum|null $navigationGroup")
                            content = content.replace("protected static ?string $navigationIcon", "protected static string|\Illuminate\Contracts\Support\Htmlable|null $navigationIcon")
                            
                            print("Restoring:", target)
                            os.makedirs(os.path.dirname(target), exist_ok=True)
                            with open(target, "w") as out:
                                out.write(content)
                            restored += 1
        except Exception as e:
            pass
print("Restored:", restored)
